Data Science Fundamentals: Python |
[Table of Contents](../index.ipynb)
- - - 
<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Exercises](./04_mod_practice.ipynb)**

# Module 1: Practice Exercises

1. Accept the user's first and last name and print them in reverse order with a space between them

In [3]:
firstname = input("Enter your first name:") # input waits for user to enter characters
lastname = input("Enter your last name:")
print(lastname + " " + firstname) #prints first and last name in reversed order

Enter your first name:Leia
Enter your last name:Fooks
Fooks Leia


2. Accept an integer (n) and compute the value of n+nn+nnn

In [11]:
n = int(input("Enter an integer:")) # int() makes sures the user enters an integer
n1 = int("%s" % n) # use % after a string to combine a string with variables
n2 = int("%s%s" % (n,n))  # % will replace %s with the string variable aka n
n3 = int("%s%s%s" % (n,n,n))
print(n1+n2+n3)

Enter an integer:5
615


3. Ask the user "What country are you from?" then print the following statement: "I have heard that [input] is a beautiful country!"

In [12]:
country = input("What country are you from?")
print("I have heard that" + " " + country + " " + "is a beautiful country!" )

What country are you from?USA
I have heard that USA is a beautiful country!


4. Write a for loop that multiplies the number 4 by all the integers from 0 to 10 and prints the following for each: "4 multiplied by [i] is [x]" 

In [14]:
n = 4
for i in range(0,11): # use for as a loop and range sets the parameters of how many times to loop
    print(n,"x",i,"=",n*i)

4 x 0 = 0
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36
4 x 10 = 40


5. Print a multiplication table that matches the following. Use two nested for loops to achieve this.

<img src="files/table.png">

In [16]:
rows = 10
columns = 10
for i in range(1,rows+1):
    for j in range(1,columns+1):
        print("{:2d}".format(i * j), end=" ") 
        # "{:2d}".format is a str.format function 
        # :d means expecting an integer
        # :2d means formats to 2 characters using padding
    print()

 1  2  3  4  5  6  7  8  9 10 
 2  4  6  8 10 12 14 16 18 20 
 3  6  9 12 15 18 21 24 27 30 
 4  8 12 16 20 24 28 32 36 40 
 5 10 15 20 25 30 35 40 45 50 
 6 12 18 24 30 36 42 48 54 60 
 7 14 21 28 35 42 49 56 63 70 
 8 16 24 32 40 48 56 64 72 80 
 9 18 27 36 45 54 63 72 81 90 
10 20 30 40 50 60 70 80 90 100 


## Additional Questions
#### 1. Is a String Immutable in Python? 
Yes

#### 2. What is the output of x AND explain why that is the output for: x = 36 / 4 * (3 +  2) * 4 + 2

In [17]:
n = 36 / 4 * (3 + 2) * 4 + 2
print(n)

182.0


The output for x is 182 because python follows order of operations (PEDMAS) for math equations.

<!--NAVIGATION-->
Module 1. [Basic Python Syntax](./01_mod_python_syntax.ipynb) | [Variables and Objects](./02_python_variables_objects.ipynb) | [Operators](./03_python_operators.ipynb) | **[Practice](./04_mod_practice.ipynb)**
<br>
[top](#)

- - -

Copyright © 2020 Qualex Consulting Services Incorporated.